bot/seasons/evergreen/reddit.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import random
+
+import discord
+from discord.ext import commands
+from discord.ext.commands.cooldowns import BucketType
+
+from bot.pagination import ImagePaginator
+
+
+log = logging.getLogger(__name__)
+
+
+class Reddit(commands.Cog):
+    """Fetches reddit posts."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    async def fetch(self, url: str) -> dict:
+        """Send a get request to the reddit API and get json response."""
+        session = self.bot.http_session
+        params = {
+            'limit': 50
+        }
+        headers = {
+            'User-Agent': 'Iceman'
+        }
+
+        async with session.get(url=url, params=params, headers=headers) as response:
+            return await response.json()
+
+    @commands.command(name='reddit')
+    @commands.cooldown(1, 10, BucketType.user)
+    async def get_reddit(self, ctx: commands.Context, subreddit: str = 'python', sort: str = "hot") -> None:
+        """
+        Fetch reddit posts by using this command.
+
+        Gets a post from r/python by default.
+        Usage:
+        --> .reddit [subreddit_name] [hot/top/new]
+        """
+        pages = []
+        sort_list = ["hot", "new", "top", "rising"]
+        if sort.lower() not in sort_list:
+            await ctx.send(f"Invalid sorting: {sort}\nUsing default sorting: `Hot`")
+            sort = "hot"
+
+        data = await self.fetch(f'https://www.reddit.com/r/{subreddit}/{sort}/.json')
+
+        try:
+            posts = data["data"]["children"]
+        except KeyError:
+            return await ctx.send('Subreddit not found!')
+        if not posts:
+            return await ctx.send('No posts available!')
+
+        if posts[1]["data"]["over_18"] is True:
+            return await ctx.send("You cannot access this Subreddit.")
```

Comment:
Describe why they can't access, e.g. "We don't allow over 18 subreddits" or something to that extent

Reply:
Keep the community clean and this feature has been discussed with the core devs and It should not allowed posts which require you to be over 18.

Reply:
We definitely want to block it, my nitpick is to make the message specific so user knows *why* it was blocked